applications/plugins/SofaSphFluid/src/SofaSphFluid/OglFluidModel.inl

```diff
@@ -368,7 +369,12 @@ void OglFluidModel<DataTypes>::drawSprites(const core::visual::VisualParams* vpa
 
     Mat4x4f matProj(fProjMat);
     Mat4x4f invmatProj;
-    invmatProj.invert(matProj);
+    if(!invmatProj.invert(matProj))
+    {
+        msg_error() << "Rendering failed (drawSprites) due to invalid input parameter \'u_projectionMatrix\' in GLSLShader";
+        this->d_componentState.setValue(sofa::core::objectmodel::ComponentState::Invalid);
+        return;
+    }
```

Comment:
I strongly suggest to not return here, as you are in the middle of offscreen FBO rendering, so it would break completely the rendering pipeline...
The better way would be:
 - computing the invmatproj before the `m_spriteNormalFBO->start();` https://github.com/sofa-framework/sofa/blob/2862461ca200d349e8033be052112538f7d11611/applications/plugins/SofaSphFluid/src/SofaSphFluid/OglFluidModel.inl#L358
 - and then testing/quitting before the FBO is started

Actually it would even be better to start to compute the matrices and test the validity right at the beginning of the drawSprites() function